Automattic/simplenote-macos
# Storage: Performance Optimizations

### Details:
While profiling #390 I've noticed few interesting details:

1. When switching from **markdown enabled** notes to **markdown disabled** ones, **Storage.theme** is being reset.
2. When we do so, we're effectivelly running `applyStyles` twice. Once manually, second round is after `endEditing` (which triggers `processEditing`)
3. Everytime the **Theme** is switched, quite a few objects are allocated over and over, including RegExes

In this PR we're:

A. Avoiding overwork: **applyStyles** won't be explicitly invoked anymore
B. Refactoring **Theme** so that switching from Markdown <> Non Markdown is ultra lightweight
C. Modernizing swift here and there.

### Test: Non Markdown
1. Add a new note
2. Ensure **Markdown** formatting is disabled
3. Type few lines of text

- [x] Verify the first line's font size is bigger than the rest of the body
- [x] Verify that increasing the font size still works as expected

### Test: Markdown
1. Add a new note
2. Ensure **Markdown** formatting is enabled
3. Enter the following text:

```
Expected to be bigger than the rest

# H1 is bold!

## H2 is bold as well

This is regular text

**Bold is Bold**

*Italics are Italics*

`Code looks monospaced and gray`

> Quotes are indented and italics

[This is a URL](www.wordpress.com)

![And this is a VERY COOL car](https://www.lantean.co/wp-content/uploads/2019/09/IMG_3753.jpeg)


```

- [x] Verify that in **Edition Mode** the first line's font size is bigger
- [x] Verify that in **Edition Mode** each one of the lines in the snippet looks as expected
- [x] Verify that in **Markdown Preview Mode** the document is properly rendered!
- [x] Verify that increasing the font size still works as expected

### Release
These changes do not require release notes.
